verifier/src/aux_verifier.rs

```diff
@@ -0,0 +1,53 @@
+use alloc::{string::ToString, vec::Vec};
+use crypto::{ElementHasher, RandomCoin, RandomCoinError};
+use math::FieldElement;
+
+/// A trait for generating the random elements required for constructing the auxiliary trace.
+pub trait AuxRandElementsGenerator {
```

Comment:
Why is this trait defined in the verifier? Shouldn't both prover and verifier use the same logic for computing random elements?

Reply:
Right, I ended up reimplementing this logic in the prover by hand. Changed it so that `AuxTraceBuilder` now uses a `AuxRandElementsGenerator`

Reply:
Correction: reverted those changes. I had forgotten why I had made a verifier-only trait, and @Al-Kindi-0 reminded me: this one needs to potentially verify the auxiliary proof as well.

Renamed `AuxTraceVerifier`, and now optionally verifies an `AuxProof`.